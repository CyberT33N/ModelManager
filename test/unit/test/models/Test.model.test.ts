/*
███████████████████████████████████████████████████████████████████████████████
██******************** PRESENTED BY t33n Software ***************************██
██                                                                           ██
██                  ████████╗██████╗ ██████╗ ███╗   ██╗                      ██
██                  ╚══██╔══╝╚════██╗╚════██╗████╗  ██║                      ██
██                     ██║    █████╔╝ █████╔╝██╔██╗ ██║                      ██
██                     ██║    ╚═══██╗ ╚═══██╗██║╚██╗██║                      ██
██                     ██║   ██████╔╝██████╔╝██║ ╚████║                      ██
██                     ╚═╝   ╚═════╝ ╚═════╝ ╚═╝  ╚═══╝                      ██
██                                                                           ██
███████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████
*/

// ==== DEPENDENCIES ====
import {
    describe, it, expect, beforeAll
} from 'vitest'

// ==== INTERNAL ====
import type { IModelCore } from '@/src/ModelManager'

describe('[UNIT TEST] - src/ModelUtils.ts', () => {
    let modelCoreDetails: IModelCore
     
    beforeAll(async() => {
        const { modelName, dbName, schema }: IModelCore = await import('@/test/models/Test.model.mjs')
  
        modelCoreDetails = {
            modelName,
            dbName,
            schema
        }
    })

    describe('[SNAPSHOT]', () => {
        it('should match the schema snapshot', () => {
            expect(modelCoreDetails.schema).toMatchSnapshot()
        })
    })
})