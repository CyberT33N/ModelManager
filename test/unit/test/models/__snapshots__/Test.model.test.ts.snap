// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[UNIT TEST] - src/ModelUtils.ts > [SNAPSHOT] > should match the schema snapshot 1`] = `
{
  "decimals": {
    "required": true,
    "type": [Function],
  },
  "name": {
    "index": true,
    "required": true,
    "type": [Function],
    "unique": true,
  },
}
`;
